### PR TITLE
Features/allow existing subnet

### PR DESCRIPTION
**Pull request by yves-vogl**
## Description

This adds the ability to specify an existing subnet. Sometimes this is created forehand.

## Type of Change

- [ ] Non-module change (e.g. CI/CD, documentation, etc.)
- [ ] Azure Verified Module updates:
  - [ ] Bugfix containing backwards compatible bug fixes
    - [ ] Someone has opened a bug report issue, and I have included "Closes #{bug_report_issue_number}" in the PR description.
    - [ ] The bug was found by the module author, and no one has opened an issue to report it yet.
  - [X] Feature update backwards compatible feature updates.
  - [ ] Breaking changes.
  - [ ] Update to documentation

# Checklist

- [X] I'm sure there are no other open Pull Requests for the same update/change
- [X] My corresponding pipelines / checks run clean and green without any errors or warnings
- [X] I did run all  [pre-commit](https://azure.github.io/Azure-Verified-Modules/contributing/terraform/terraform-contribution-flow/#5-run-pre-commit-checks) checks

### CODE REVIEW

**Comment by nellyk** [comment]
@yves-vogl  Thank you for your contributions.

We will review the PRs and get back to you!

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -0,0 +1,20 @@
+# To contribute improvements to CI/CD templates, please follow the Development guide at:
+# https://docs.gitlab.com/ee/development/cicd/templates.html
+# This specific template is located at:
+# https://gitlab.com/gitlab-org/gitlab/-/blob/master/lib/gitlab/ci/templates/Terraform-Module.gitlab-ci.yml
+
+include:
```
Comment (nellyk) [inline .gitlab-ci.yml:6]: The pipelines are currently running on Github actions. Could you please remove this?

Comment (nellyk) [reply .gitlab-ci.yml:6]: I'll kick of the pipeline for this changes

Comment (yves-vogl) [reply .gitlab-ci.yml:6]: I did this in a hurry. I'll finish my work a submit a larger PR with proper care.

Comment (nellyk) [reply .gitlab-ci.yml:6]: @yves-vogl no worries let us know if you have any questions

**Pull request closed without merging**